### PR TITLE
Add .SortedMergeBy()

### LINKED ISSUE #24: Add .SortedMergeBy()

### PR DESCRIPTION
**Pull request by viceroypenguin**
Fixes #24

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -210,6 +210,7 @@ dotnet_code_quality.ca1711.allowed_suffixes = Ex
 dotnet_diagnostic.CS0105.severity = error               # CS0105: Using directive is unnecessary.
 dotnet_diagnostic.CS1573.severity = error               # CS1573: Missing XML comment for parameter
 dotnet_diagnostic.CS1591.severity = error               # CS1591: Missing XML comment for publicly visible type or member
+dotnet_diagnostic.CS1712.severity = error				# CS1712: Type parameter has no matching typeparam tag in the XML comment (but other type parameters do)
```
Comment (Arithmomaniac) [inline .editorconfig:213]: Is this indent misaligned, or is it just the GH viewer?

File: Source/SuperLinq/readme.md [hunk 2]
```diff
@@ -584,13 +584,27 @@ descending) into a single sequence that preserves that order.
...
 ### SortedMergeDescending
 
 Merges two or more sequences that are in a common order (either ascending or
 descending) into a single sequence that preserves that order.
 
 This method has 2 overloads.
 
+### SortedMergeByDescending
+
+Merges two or more sequences that are in a common order (either ascending or descending)
```
Comment (Arithmomaniac) [inline Source/SuperLinq/readme.md:603]: This "either" isn't true for the *Descending methods

File: Tests/SuperLinq.Test/SortedMergeByTest.cs [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+	{
+		using var sequenceA = TestingSequence.Of<int>();
+
+		// Expected and thrown by BreakingSequence
+		Assert.Throws<InvalidOperationException>(() =>
+			sequenceA.SortedMergeBy(SuperEnumerable.Identity<int>(), OrderByDirection.Ascending, new BreakingSequence<int>()).Consume());
+	}
+
+	/// <summary>
+	/// Verify that SortedMergeBy throws an exception if invoked on a <see langword="null"/> sequence.
```
Comment (Arithmomaniac) [inline Tests/SuperLinq.Test/SortedMergeByTest.cs:35]: Is there a backlog item to flip on nullable annotations?

File: Source/SuperLinq/SortedMerge.cs [hunk 4]
```diff
@@ -173,19 +189,163 @@ public static IEnumerable<TSource> SortedMerge<TSource>(this IEnumerable<TSource
...
+	{
+		return SortedMergeBy(source, Identity<TSource>(), direction, comparer, otherSequences);
+	}
+
+	/// <summary>
+	/// Merges two or more sequences that are in a common order (either ascending or descending)
+	/// according to a key into a single sequence that preserves that order.
+	/// </summary>
+	/// <remarks>
+	/// Using SortedMerge on sequences that are not ordered or are not in the same order produces
```
Comment (Arithmomaniac) [inline Source/SuperLinq/SortedMerge.cs:201]: SortedMergeBy

**Review by Arithmomaniac — APPROVED** [review]

File: .editorconfig (hunk 1 above)
Comment (viceroypenguin) [reply .editorconfig]: Corrected

File: Source/SuperLinq/readme.md (hunk 2 above)
Comment (viceroypenguin) [reply Source/SuperLinq/readme.md]: Corrected

File: Source/SuperLinq/SortedMerge.cs (hunk 4 above)
Comment (viceroypenguin) [reply Source/SuperLinq/SortedMerge.cs]: Corrected.

**Pull request merged**